File: domain/src/main/java/com/blackmirror/dongda/domain/Interactor/SendSmsUseCase.java
package com.blackmirror.dongda.domain.Interactor;

import com.blackmirror.dongda.domain.model.SendSmsBean;
import com.blackmirror.dongda.domain.repository.LoginRepository;

import io.reactivex.Observable;

public class SendSmsUseCase extends LoginUseCase<SendSmsBean> {

    private final LoginRepository repository;

    public SendSmsUseCase(LoginRepository repository) {
        this.repository = repository;
    }


    public Observable<SendSmsBean> sendSms(String phone) {
        return repository.sendSms(phone);
    }
}

File: domain/src/main/java/com/blackmirror/dongda/domain/Interactor/login/SendSmsUseCase.kt
package com.blackmirror.dongda.domain.Interactor.login

import com.blackmirror.dongda.domain.model.SendSmsBean
import com.blackmirror.dongda.domain.repository.LoginRepository
import io.reactivex.Observable

class SendSmsUseCase(private val repository: LoginRepository){


    fun sendSms(phone: String): Observable<SendSmsBean> {
        return repository.sendSms(phone)
    }
}
